### PR TITLE
Correct some spelling errors (continued)

### PR DESCRIPTION
**Pull request by herbelin**
This is a rebase of #8117 (+ another spelling fix in a doc label). (I did not succeed to force-push to #8117.)

**Kind:** infrastructure

Lintian found some spelling errors in the Debian packaging for coq. Fix them most places they appear in the current source. (Don’t change documentation anchor names, as that would invalidate external deeplinks.)

This also fixes a bug in coqdoc: prior to this commit, coqdoc would highlight `instanciate` but not `instantiate`.

### CODE REVIEW

**Review by gares — APPROVED** [review]
Thanks!

File: doc/sphinx/proof-engine/tactics.rst [hunk 1]
```diff
@@ -103,7 +103,7 @@ bindings_list`` where ``bindings_list`` may be of two different forms:
 
   .. exn:: Not the right number of missing arguments.
 
-.. _occurencessets:
+.. _occurrencessets:
```
Comment (Zimmi48) [inline doc/sphinx/proof-engine/tactics.rst]: This will change the name of the anchor, but I still support fixing it.

**Review by Zimmi48 — APPROVED** [review]

**Review by ejgallego — APPROVED** [review]
OK with my part, thanks!

**Review by ppedrot — APPROVED** [review]

**Review by maximedenes — APPROVED** [review]
`clambda` change ok

**Review by silene — APPROVED** [review]

**Comment by herbelin** [comment]
Not clear who would be the most canonical assignee on this multi-component PR? @ppedrot, would you like to be the assignee?

**Comment by ejgallego** [comment]
> Not clear who would be the most canonical assignee on this multi-component PR? @ppedrot, would you like to be the assignee?

This is trivial enough I can take it.

**Comment by ejgallego** [comment]
CI failures due to #8704 overlays not having been merged, merging.

**Pull request merged**